installer/pkg/common/storage.go

```diff
@@ -120,14 +117,10 @@ func AddStorageMounts(ctx *RenderContext, pod *corev1.PodSpec, container ...stri
 		return nil
 	}
 
-	if ctx.Config.ObjectStorage.S3 != nil {
-		return nil
-	}
-
-	if pointer.BoolDeref(ctx.Config.ObjectStorage.InCluster, false) {
+	if useMinio(ctx) {
 		// builtin storage needs no extra mounts
```

Comment:
Minio can be in two states - internal storage and gateway to external. Because the interface is the same, we don't actually need to know how we're doing it, just that we are doing it.

This is how it seems to work with Helm (so what I've replicated in Installer), that the storage is either GCP or Minio, regardless of whether Minio is internal or gateway.

In the future (#6777), there will be an additional "if s3" in the `useMinio` function. As there are two places where we need to know if we're using Minio (and not _how_ we're using it), it seemed sensible to have a function to work it out

Reply:
That makes a total sense. I was just a tad confused by the comment in this line.